src/metabase/lib/js.cljs

```diff
@@ -465,6 +469,82 @@
                 convert/->pMBQL)]
     (lib.core/find-visible-column-for-ref a-query stage-number ref)))
 
+;; TODO: Added as an expedient to fix metabase/metabase#32373. Due to the interaction with viz-settings, this issue
+;; was difficult to fix entirely within MLv2. Once viz-settings are ported, this function should not be needed, and the
+;; FE logic using it should be ported to MLv2 behind more meaningful names.
+(defn ^:export visible-columns
+  "Return a sequence of column metadatas for columns visible at the given stage of the query. Takes a map of options
+  which defines the set of columns to return. All the options default to *true*, ie. everything is returned by default.
+  ```
+  {
+    includeJoined: true,
+    includeExpressions true,
+    includeImplicitlyJoinable true,
+  }
+  ```"
+  [a-query stage-number ^js js-options]
+  (let [^js js-options (or js-options #js {})
+        options        {:include-joined?              (gobj/get js-options "includeJoined" true)
+                        :include-expressions?         (gobj/get js-options "includeExpressions" true)
+                        :include-implicitly-joinable? (gobj/get js-options "includeImplicitlyJoinable" true)}
```

Comment:
would this be easier if we just called `js->clj` on the JS object and then `update-keys` with `u/-:kebab-case-en` and passed that map in directly? Main reason I'm asking is because if we add a new option we'd get it for free that way and not need to hardcode it here as well

Reply:
These options are now never passed as anything but `{}` by the FE, so I've just dropped the argument and all the conversion code.